haystack/components/rankers/sentence_transformers_diversity.py

```diff
@@ -88,15 +162,50 @@ def __init__(
         self.device = ComponentDevice.resolve_device(device)
         self.token = token
         self.model = None
-        if similarity not in ["dot_product", "cosine"]:
-            raise ValueError(f"Similarity must be one of 'dot_product' or 'cosine', but got {similarity}.")
-        self.similarity = similarity
+        self.similarity = self._parse_similarity(similarity)
         self.query_prefix = query_prefix
         self.document_prefix = document_prefix
         self.query_suffix = query_suffix
         self.document_suffix = document_suffix
         self.meta_fields_to_embed = meta_fields_to_embed or []
         self.embedding_separator = embedding_separator
+        self.strategy = self._parse_strategy(strategy)
+        self._check_lambda_threshold(lambda_threshold, strategy)  # type: ignore
+        self.lambda_threshold = lambda_threshold or 0.5
+
+    @staticmethod
+    def _parse_similarity(similarity: Union[str, Similarity]) -> "Similarity":
```

Comment:
This functions is only ever called in the init method and can therefore be simplified to:
```python
if isinstance(similarity, str):
	similarity = Similarity.from_str(similarity)

self.similarity = similarity
```